File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
# 프로그래머스 42626 더 맵게 LV.2
import heapq


def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while scoville:
        if len(scoville) == 1 and scoville[-1] < K:
            return -1
        if all(sc >= K for sc in scoville):
            return answer
        else:
            heapq.heappush(scoville, heapq.heappop(scoville) + (heapq.heappop(scoville) * 2))
            answer += 1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        Queue<Integer> pq = new PriorityQueue<>(Comparator.naturalOrder());
        for (int s : scoville) pq.add(s);
        
        int answer = 0;
        while (pq.peek() < K) {
            if (pq.size() == 1) break;
            int f = pq.poll(), s = pq.poll();
            pq.add(f + (s * 2));
            answer++;
        }
        
        return pq.peek() < K ? -1 : answer;
    }
}